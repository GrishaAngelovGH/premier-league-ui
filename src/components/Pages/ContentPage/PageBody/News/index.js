import News, { newsItems } from './News'

export { newsItems }

export default News