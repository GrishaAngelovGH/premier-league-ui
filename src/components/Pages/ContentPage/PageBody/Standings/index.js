import Standings from './Standings'
import standingItems from './standingsItems'

export { standingItems }

export default Standings