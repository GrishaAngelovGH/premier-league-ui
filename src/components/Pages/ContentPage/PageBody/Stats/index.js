import Stats from './Stats'
import statsItems from './statsItems'

export { statsItems }

export default Stats