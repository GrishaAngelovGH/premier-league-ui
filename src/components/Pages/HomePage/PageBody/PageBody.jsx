import Timetable from './Timetable'

const PageBody = ({ content }) => (
	<Timetable content={content} />
)

export default PageBody