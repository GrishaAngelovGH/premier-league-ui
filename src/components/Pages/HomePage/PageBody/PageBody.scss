.row-item {
    cursor: pointer;
    &:hover {
        background-color: #e2e2ef;
    }
}
