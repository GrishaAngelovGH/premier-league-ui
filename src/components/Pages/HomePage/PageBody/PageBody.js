import PropTypes from 'prop-types'

import images from 'images'
import './PageBody.scss'

const PageBody = ({ content }) => (
    <div className='row no-gutters'>
        <div className='col-md-12'>
            {
                content.map((v, i) => (
                    <div key={i} className='row no-gutters'>
                        {
                            v.map((item => (
                                <div key={item.id} className='col-md-6 border'>
                                    <div className='row no-gutters row-item'>
                                        <div className='col-8 col-md-8 p-2'>
                                            <div className='row justify-content-center m-1'>
                                                <div className='col-2 col-md-2'>
                                                    <img src={images[item.firstTeam.image]} width={25} />
                                                </div>
                                                <div className='col-8 col-md-8'>
                                                    {item.firstTeam.name}
                                                </div>
                                                <div className='col-1 col-md-1'>
                                                    {item.firstTeam.score >= 0 ? item.firstTeam.score : ''}
                                                </div>
                                            </div>

                                            <div className='row justify-content-center m-1'>
                                                <div className='col-2 col-md-2'>
                                                    <img src={images[item.secondTeam.image]} width={25} />
                                                </div>
                                                <div className='col-8 col-md-8'>
                                                    {item.secondTeam.name}
                                                </div>
                                                <div className='col-1 col-md-1'>
                                                    {item.secondTeam.score >= 0 ? item.secondTeam.score : ''}
                                                </div>
                                            </div>
                                        </div>

                                        <div className='col-4 col-md-4 d-flex flex-column justify-content-center text-center'>
                                            <div>{item.day}</div>
                                            <div>{item.time}</div>
                                        </div>
                                    </div>
                                </div>
                            )))
                        }
                    </div>
                ))
            }
        </div>
    </div>
)

PageBody.propTypes = {
    content: PropTypes.array.isRequired
}

export default PageBody