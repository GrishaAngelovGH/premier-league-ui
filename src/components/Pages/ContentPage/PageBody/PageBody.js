
import { Switch, Route, Redirect } from 'react-router-dom'

import Matches from './Matches'
import News, { newsItems } from './News'
import Standings, { standingItems } from './Standings'
import Stats from './Stats'
import Players from './Players'

const PageBody = () => (
    <Switch>
        <Route exact path='/content'>
            <Redirect to='/content/matches' />
        </Route>
        <Route path='/content/matches' children={<Matches />} />
        <Route path='/content/news' children={<News newsItems={newsItems} />} />
        <Route path='/content/standings' children={<Standings standingItems={standingItems} />} />
        <Route path='/content/stats' children={<Stats />} />
        <Route path='/content/players' children={<Players />} />
    </Switch>
)

export default PageBody