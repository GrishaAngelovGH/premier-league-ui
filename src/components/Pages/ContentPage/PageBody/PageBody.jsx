
import { Routes, Route, Navigate } from 'react-router-dom'

import Matches from './Matches'
import News from './News'
import Standings, { standingItems } from './Standings'
import Stats, { statsItems } from './Stats'
import Players from './Players'

const PageBody = () => (
	<Routes>
		<Route path='/' element={<Navigate to='/content/matches' />} />
		<Route path='/matches' element={<Matches />} />
		<Route path='/news' element={<News />} />
		<Route path='/standings' element={<Standings standingItems={standingItems} />} />
		<Route path='/stats' element={<Stats statsItems={statsItems} />} />
		<Route path='/players' element={<Players />} />
	</Routes>
)

export default PageBody