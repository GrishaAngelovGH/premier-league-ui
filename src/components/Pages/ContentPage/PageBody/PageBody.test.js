import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import PageBody from './PageBody'

test('should render ContentPage/PageBody component', () => {
    const view = render(
        <MemoryRouter initialEntries={['/matches']}>
            <PageBody />
        </MemoryRouter>
    )

    expect(view).toMatchSnapshot()
})