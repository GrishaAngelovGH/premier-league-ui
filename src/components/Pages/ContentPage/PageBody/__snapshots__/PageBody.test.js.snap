// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ContentPage/PageBody component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row no-gutters"
      >
        <div
          class="col-md-12"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-md-6 border"
            >
              <div
                class="row no-gutters row-item"
              >
                <div
                  class="col-8 col-md-8 p-2"
                >
                  <div
                    class="row justify-content-center m-1"
                  >
                    <div
                      class="col-2 col-md-2"
                    >
                      <img
                        src="crystal-palace.png"
                        width="25"
                      />
                    </div>
                    <div
                      class="col-8 col-md-8"
                    >
                      Crystal Palace
                    </div>
                    <div
                      class="col-1 col-md-1"
                    >
                      0
                    </div>
                  </div>
                  <div
                    class="row justify-content-center m-1"
                  >
                    <div
                      class="col-2 col-md-2"
                    >
                      <img
                        src="newcastle.png"
                        width="25"
                      />
                    </div>
                    <div
                      class="col-8 col-md-8"
                    >
                      Newcastle
                    </div>
                    <div
                      class="col-1 col-md-1"
                    >
                      2
                    </div>
                  </div>
                </div>
                <div
                  class="col-4 col-md-4 d-flex flex-column justify-content-center text-center"
                >
                  <div>
                    FT
                  </div>
                  <div>
                    Yesterday
                  </div>
                </div>
              </div>
            </div>
            <div
              class="col-md-6 border"
            >
              <div
                class="row no-gutters row-item"
              >
                <div
                  class="col-8 col-md-8 p-2"
                >
                  <div
                    class="row justify-content-center m-1"
                  >
                    <div
                      class="col-2 col-md-2"
                    >
                      <img
                        src="brighton.png"
                        width="25"
                      />
                    </div>
                    <div
                      class="col-8 col-md-8"
                    >
                      Brighton
                    </div>
                    <div
                      class="col-1 col-md-1"
                    >
                      1
                    </div>
                  </div>
                  <div
                    class="row justify-content-center m-1"
                  >
                    <div
                      class="col-2 col-md-2"
                    >
                      <img
                        src="liverpool.png"
                        width="25"
                      />
                    </div>
                    <div
                      class="col-8 col-md-8"
                    >
                      Liverpool
                    </div>
                    <div
                      class="col-1 col-md-1"
                    >
                      3
                    </div>
                  </div>
                </div>
                <div
                  class="col-4 col-md-4 d-flex flex-column justify-content-center text-center"
                >
                  <div>
                    FT
                  </div>
                  <div>
                    Yesterday
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row no-gutters"
          >
            <div
              class="col-md-6 border"
            >
              <div
                class="row no-gutters row-item"
              >
                <div
                  class="col-8 col-md-8 p-2"
                >
                  <div
                    class="row justify-content-center m-1"
                  >
                    <div
                      class="col-2 col-md-2"
                    >
                      <img
                        src="man-city.png"
                        width="25"
                      />
                    </div>
                    <div
                      class="col-8 col-md-8"
                    >
                      Man. City
                    </div>
                    <div
                      class="col-1 col-md-1"
                    >
                      2
                    </div>
                  </div>
                  <div
                    class="row justify-content-center m-1"
                  >
                    <div
                      class="col-2 col-md-2"
                    >
                      <img
                        src="burnley.png"
                        width="25"
                      />
                    </div>
                    <div
                      class="col-8 col-md-8"
                    >
                      Burnley
                    </div>
                    <div
                      class="col-1 col-md-1"
                    >
                      1
                    </div>
                  </div>
                </div>
                <div
                  class="col-4 col-md-4 d-flex flex-column justify-content-center text-center"
                >
                  <div>
                    FT
                  </div>
                  <div>
                    Yesterday
                  </div>
                </div>
              </div>
            </div>
            <div
              class="col-md-6 border"
            >
              <div
                class="row no-gutters row-item"
              >
                <div
                  class="col-8 col-md-8 p-2"
                >
                  <div
                    class="row justify-content-center m-1"
                  >
                    <div
                      class="col-2 col-md-2"
                    >
                      <img
                        src="everton.png"
                        width="25"
                      />
                    </div>
                    <div
                      class="col-8 col-md-8"
                    >
                      Everton
                    </div>
                    <div
                      class="col-1 col-md-1"
                    />
                  </div>
                  <div
                    class="row justify-content-center m-1"
                  >
                    <div
                      class="col-2 col-md-2"
                    >
                      <img
                        src="leeds.png"
                        width="25"
                      />
                    </div>
                    <div
                      class="col-8 col-md-8"
                    >
                      Leeds United
                    </div>
                    <div
                      class="col-1 col-md-1"
                    />
                  </div>
                </div>
                <div
                  class="col-4 col-md-4 d-flex flex-column justify-content-center text-center"
                >
                  <div>
                    Today
                  </div>
                  <div>
                    19:30
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row no-gutters"
          >
            <div
              class="col-md-6 border"
            >
              <div
                class="row no-gutters row-item"
              >
                <div
                  class="col-8 col-md-8 p-2"
                >
                  <div
                    class="row justify-content-center m-1"
                  >
                    <div
                      class="col-2 col-md-2"
                    >
                      <img
                        src="west-brom.png"
                        width="25"
                      />
                    </div>
                    <div
                      class="col-8 col-md-8"
                    >
                      West Brom
                    </div>
                    <div
                      class="col-1 col-md-1"
                    />
                  </div>
                  <div
                    class="row justify-content-center m-1"
                  >
                    <div
                      class="col-2 col-md-2"
                    >
                      <img
                        src="sheffield.png"
                        width="25"
                      />
                    </div>
                    <div
                      class="col-8 col-md-8"
                    >
                      Sheffield United
                    </div>
                    <div
                      class="col-1 col-md-1"
                    />
                  </div>
                </div>
                <div
                  class="col-4 col-md-4 d-flex flex-column justify-content-center text-center"
                >
                  <div>
                    Today
                  </div>
                  <div>
                    22:00
                  </div>
                </div>
              </div>
            </div>
            <div
              class="col-md-6 border"
            >
              <div
                class="row no-gutters row-item"
              >
                <div
                  class="col-8 col-md-8 p-2"
                >
                  <div
                    class="row justify-content-center m-1"
                  >
                    <div
                      class="col-2 col-md-2"
                    >
                      <img
                        src="southampton.png"
                        width="25"
                      />
                    </div>
                    <div
                      class="col-8 col-md-8"
                    >
                      Southampton
                    </div>
                    <div
                      class="col-1 col-md-1"
                    />
                  </div>
                  <div
                    class="row justify-content-center m-1"
                  >
                    <div
                      class="col-2 col-md-2"
                    >
                      <img
                        src="man-utd.png"
                        width="25"
                      />
                    </div>
                    <div
                      class="col-8 col-md-8"
                    >
                      Man United
                    </div>
                    <div
                      class="col-1 col-md-1"
                    />
                  </div>
                </div>
                <div
                  class="col-4 col-md-4 d-flex flex-column justify-content-center text-center"
                >
                  <div>
                    Tomorrow
                  </div>
                  <div>
                    16:00
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row no-gutters"
    >
      <div
        class="col-md-12"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-md-6 border"
          >
            <div
              class="row no-gutters row-item"
            >
              <div
                class="col-8 col-md-8 p-2"
              >
                <div
                  class="row justify-content-center m-1"
                >
                  <div
                    class="col-2 col-md-2"
                  >
                    <img
                      src="crystal-palace.png"
                      width="25"
                    />
                  </div>
                  <div
                    class="col-8 col-md-8"
                  >
                    Crystal Palace
                  </div>
                  <div
                    class="col-1 col-md-1"
                  >
                    0
                  </div>
                </div>
                <div
                  class="row justify-content-center m-1"
                >
                  <div
                    class="col-2 col-md-2"
                  >
                    <img
                      src="newcastle.png"
                      width="25"
                    />
                  </div>
                  <div
                    class="col-8 col-md-8"
                  >
                    Newcastle
                  </div>
                  <div
                    class="col-1 col-md-1"
                  >
                    2
                  </div>
                </div>
              </div>
              <div
                class="col-4 col-md-4 d-flex flex-column justify-content-center text-center"
              >
                <div>
                  FT
                </div>
                <div>
                  Yesterday
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-md-6 border"
          >
            <div
              class="row no-gutters row-item"
            >
              <div
                class="col-8 col-md-8 p-2"
              >
                <div
                  class="row justify-content-center m-1"
                >
                  <div
                    class="col-2 col-md-2"
                  >
                    <img
                      src="brighton.png"
                      width="25"
                    />
                  </div>
                  <div
                    class="col-8 col-md-8"
                  >
                    Brighton
                  </div>
                  <div
                    class="col-1 col-md-1"
                  >
                    1
                  </div>
                </div>
                <div
                  class="row justify-content-center m-1"
                >
                  <div
                    class="col-2 col-md-2"
                  >
                    <img
                      src="liverpool.png"
                      width="25"
                    />
                  </div>
                  <div
                    class="col-8 col-md-8"
                  >
                    Liverpool
                  </div>
                  <div
                    class="col-1 col-md-1"
                  >
                    3
                  </div>
                </div>
              </div>
              <div
                class="col-4 col-md-4 d-flex flex-column justify-content-center text-center"
              >
                <div>
                  FT
                </div>
                <div>
                  Yesterday
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="row no-gutters"
        >
          <div
            class="col-md-6 border"
          >
            <div
              class="row no-gutters row-item"
            >
              <div
                class="col-8 col-md-8 p-2"
              >
                <div
                  class="row justify-content-center m-1"
                >
                  <div
                    class="col-2 col-md-2"
                  >
                    <img
                      src="man-city.png"
                      width="25"
                    />
                  </div>
                  <div
                    class="col-8 col-md-8"
                  >
                    Man. City
                  </div>
                  <div
                    class="col-1 col-md-1"
                  >
                    2
                  </div>
                </div>
                <div
                  class="row justify-content-center m-1"
                >
                  <div
                    class="col-2 col-md-2"
                  >
                    <img
                      src="burnley.png"
                      width="25"
                    />
                  </div>
                  <div
                    class="col-8 col-md-8"
                  >
                    Burnley
                  </div>
                  <div
                    class="col-1 col-md-1"
                  >
                    1
                  </div>
                </div>
              </div>
              <div
                class="col-4 col-md-4 d-flex flex-column justify-content-center text-center"
              >
                <div>
                  FT
                </div>
                <div>
                  Yesterday
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-md-6 border"
          >
            <div
              class="row no-gutters row-item"
            >
              <div
                class="col-8 col-md-8 p-2"
              >
                <div
                  class="row justify-content-center m-1"
                >
                  <div
                    class="col-2 col-md-2"
                  >
                    <img
                      src="everton.png"
                      width="25"
                    />
                  </div>
                  <div
                    class="col-8 col-md-8"
                  >
                    Everton
                  </div>
                  <div
                    class="col-1 col-md-1"
                  />
                </div>
                <div
                  class="row justify-content-center m-1"
                >
                  <div
                    class="col-2 col-md-2"
                  >
                    <img
                      src="leeds.png"
                      width="25"
                    />
                  </div>
                  <div
                    class="col-8 col-md-8"
                  >
                    Leeds United
                  </div>
                  <div
                    class="col-1 col-md-1"
                  />
                </div>
              </div>
              <div
                class="col-4 col-md-4 d-flex flex-column justify-content-center text-center"
              >
                <div>
                  Today
                </div>
                <div>
                  19:30
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="row no-gutters"
        >
          <div
            class="col-md-6 border"
          >
            <div
              class="row no-gutters row-item"
            >
              <div
                class="col-8 col-md-8 p-2"
              >
                <div
                  class="row justify-content-center m-1"
                >
                  <div
                    class="col-2 col-md-2"
                  >
                    <img
                      src="west-brom.png"
                      width="25"
                    />
                  </div>
                  <div
                    class="col-8 col-md-8"
                  >
                    West Brom
                  </div>
                  <div
                    class="col-1 col-md-1"
                  />
                </div>
                <div
                  class="row justify-content-center m-1"
                >
                  <div
                    class="col-2 col-md-2"
                  >
                    <img
                      src="sheffield.png"
                      width="25"
                    />
                  </div>
                  <div
                    class="col-8 col-md-8"
                  >
                    Sheffield United
                  </div>
                  <div
                    class="col-1 col-md-1"
                  />
                </div>
              </div>
              <div
                class="col-4 col-md-4 d-flex flex-column justify-content-center text-center"
              >
                <div>
                  Today
                </div>
                <div>
                  22:00
                </div>
              </div>
            </div>
          </div>
          <div
            class="col-md-6 border"
          >
            <div
              class="row no-gutters row-item"
            >
              <div
                class="col-8 col-md-8 p-2"
              >
                <div
                  class="row justify-content-center m-1"
                >
                  <div
                    class="col-2 col-md-2"
                  >
                    <img
                      src="southampton.png"
                      width="25"
                    />
                  </div>
                  <div
                    class="col-8 col-md-8"
                  >
                    Southampton
                  </div>
                  <div
                    class="col-1 col-md-1"
                  />
                </div>
                <div
                  class="row justify-content-center m-1"
                >
                  <div
                    class="col-2 col-md-2"
                  >
                    <img
                      src="man-utd.png"
                      width="25"
                    />
                  </div>
                  <div
                    class="col-8 col-md-8"
                  >
                    Man United
                  </div>
                  <div
                    class="col-1 col-md-1"
                  />
                </div>
              </div>
              <div
                class="col-4 col-md-4 d-flex flex-column justify-content-center text-center"
              >
                <div>
                  Tomorrow
                </div>
                <div>
                  16:00
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
